File: net/smart-recommendations-api/TrackVisitor.cs
// DocSection: rapi_v2_track_visitor
using Kentico.Kontent.Recommendations;
using Kentico.Kontent.Recommendations.Models;

// Creates an instance of the recommendation client
var recommendationClient = new RecommendationClient(accessToken: "<YOUR_RECOMMENDATION_API_KEY>", timeoutSeconds: 5);

var visitor = new VisitorDetails {
    // Provide IP if you enabled Geolocation
    Ip = "192.166.12.4",
    Referer = "google.com",
    // Provide Location if you disabled Geolocation
    Location = new LocationDetails {
        City = "Brno",
        Country = "Czech Republic",
        Timezone = "CET"
    }
};

// Creates a new recommendation request
var recommendationRequest = new RecommendationRequest {
    VisitId = "visitorId123",
    CurrentItemCodename = "example_blogpost",
    ResponseLimit = 3,
    RequestedTypeCodename = "blog_post",
    RecommendationSettings = recommendationSettings,
    Visitor = visitor
};

// Returns the requested number of recommended content items (their codenames)
RecommendedContentItem[] recommendedArticles = await recommendationClient.GetRecommendationsAsync(recommendationRequest);
// EndDocSection

File: js/smart-recommendations-api/TrackVisitor.js
// DocSection: rapi_v2_track_visitor
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
import { RecommendationClient } from '@kentico/kontent-recommendations';

// Creates an instance of the recommendation client
const client = new RecommendationClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_SMART_RECOMMENDATION_API_KEY>'
});

// Creates a visitor profile in Smart Recommendations
await client.trackVisitor()
  .withData({
    visitId: 'visitorId123',
    visitor: {
      ip: '192.166.12.4',
      referer: 'https://www.google.com/search?hl=en&q=kentico%20kontent',
      location: {
        city: 'Brno',
        country: 'Czech Republic',
        timezone: 'CET'
      },
      custom: {
        persona: 'developer'
      }
    }
  }).toPromise();
// EndDocSection
